### ISSUE TITLE
Long build time (probably building gcc10 from scratch)

### ISSUE THREAD

**Issue by Ali-Flt** [bartoszek/AUR-colmap#4]
This (and colmap-git) takes 5~6 hours to build. (I used yay). I think it is building gcc10 from scratch.

**Comment by bartoszek** [comment]
Lol, share your system spec, please.

**Comment by Ali-Flt** [comment]
What do you mean by lol? Is it normal for this to build gcc from scratch? 
- 12th Gen Intel(R) Core(TM) i5-12600KF
- 32GB DDR4
- Yes I have a SSD 

(Not sure if this information is relevant to the issue though.)

I cloned colmap's git and built it from there and it took less than 5 minutes so there is definitely something wrong here.
I told you that your package is building gcc10 and you said lol and asked for my system spec instead of asking for the build logs or anything that may help in finding the issue which is quite weird. 
I'm confident that this is building gcc so maybe think of why that could be. My arch was up to date at the time of trying to install colmap btw.

**Comment by bartoszek** [comment]
Weird... gcc-10 isn't part of dependencies chain for colmap. I think your wrong, perhaps metis build which use `gnu-make` runs in single thread? What's your makepkg.conf MAKEFLAGS?

**Comment by Ali-Flt** [comment]
where should I look for the makepkg.conf? I have not added any additional MAKEFLAGS manually if that's what you are asking. I only ran `yay -S colmap`.

As you can see in the output of `yay -S colmap` it is looking for gcc10 (and it will try to build it if it is not installed). 
![image](https://github.com/bartoszek/AUR-colmap/assets/60854595/47e8006a-d57f-40a7-a674-10b50104a839)

Do you know why this is?

**Comment by bartoszek** [comment]
Now I see (－_ლ) yes gcc:10 is host compiler for cuda:11. But this has to be built only once, and even on my old 4790k it won't take more than 2h.

**Comment by Ali-Flt** [comment]
Sorry for the late reply. Thanks for the tip on using parallel compilation, I will try it on my next builds. But nonetheless, I don't think it is sane for colmap to build gcc10 from scratch. Even if it takes 2 hours that's a really long time to build a package not to mention the fact that building gcc is very harmful to SSDs due to its significant amount of disk usage.

It would be great if there was a way to select between building gcc10 or downloading its binary instead. I tried installing gcc10 from pacman but found only gcc 12 and gcc 13 available. 

As a workaround, it would be great if you could add a colmap-bin package that doesn't require building gcc:10 or cuda:11 from scratch.

**Comment by bartoszek** [comment]
Well, it would be a solution if `colmap` provided binaries for Linux.

Best advice I can give is:
* build gcc/cuda once and save them in a local repo.
* build inside chroot container with repo pinned to a date when `gcc/cuda` version was available.
* use `downgrade` to get old `gcc/cuda` from Arch Archive to build `colmap` with.

I'm personally using combination 1-3: download old version with `downgrade`, add to local repo called `old`

```sh
pacman -Ss ^cuda$
extra/cuda 12.3.2-1
    NVIDIA's GPU programming toolkit
old/cuda 11.8.0-1
    NVIDIA's GPU programming toolkit
```

Let's see if `colmap:3.9.1` is compatible with `cuda:12`  which will render this conversation redundant 😊

**Comment by bartoszek** [comment]
Fixed in 2ac6207058f17676307509a7e187e5c78ed7d1d7
